adi/ltc2983.py

```diff
@@ -0,0 +1,79 @@
+import numbers
+from collections import OrderedDict
+from collections.abc import Iterable
+
+import numpy as np
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class ltc2983(rx, context_manager):
+    """ LTC2983 Multi-Sensor Temperature Measurement System """
+
+    _complex_data = False
+    channel = None
+    _device_name = ""
+    _rx_data_type = np.int32
+
+    def __init__(self, uri=""):
+        context_manager.__init__(self, uri, self._device_name)
+        self._ctrl = self._ctx.find_device("ltc2983")
+        self._rxadc = self._ctx.find_device("ltc2983")
+
+        # dynamically get channels
+        _channels = []
+        for ch in self._ctrl.channels:
+            self._rx_channel_names.append(ch.id)
+            _channels.append((ch.id, self._channel(self._ctrl, ch)))
+        self.channel = OrderedDict(_channels)
+
+        rx.__init__(self)
+
+    class _channel(attribute):
+        """ LTC2983 channel """
+
+        def __init__(self, ctrl, channel):
+            self._ctrl = ctrl
+            self._channel = channel
+
+            self.name = channel.id
+
+            # raw value attribute is '<x>_raw' with
+            # <x>=thermistor,rtd,diode,thermocouple,direct_adc
+            raw_attr_name = [x for x in channel.attrs.keys() if x.endswith("raw")]
+            assert len(raw_attr_name) == 1
```

Comment:
Can you add an error message to the assert

Reply:
Removed this. I was working against an older revision of the kernel driver. The newest version analogdevicesinc/linux@2b453a1 fixes this with a simpler attribute naming scheme.